FTCBotCode/Competition24_25.java

```diff
@@ -1,25 +1,41 @@
 package org.firstinspires.ftc.teamcode;
 
-// import com.qualcomm.robotcore.hardware.hardwareMap;
 import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
 import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
 import com.qualcomm.robotcore.hardware.DcMotor;
 import com.qualcomm.robotcore.hardware.DcMotorSimple;
+import com.qualcomm.robotcore.hardware.Servo;
+import com.qualcomm.robotcore.hardware.ServoController;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
 
 @TeleOp(name = "FTC Competition (Blocks to Java)")
 public class Competition24_25 extends LinearOpMode {
-
+  //Drive
   private DcMotor frontRight;
   private DcMotor backRight;
   private DcMotor frontLeft;
   private DcMotor backLeft;
 
+  //Arm
+  private ServoController ControlHub_ServoController;
+  private Servo pincher;
+  private Servo wrist;
+  private DcMotor elbow;
+  private DcMotor lift
+  private DcMotor shoulder
+
   public void runOpMode() {
+    //Drive
     float horizontal;
     float vertical;
     float pivot;
-    float armDown;
-    float armUp;
+    double drivePower = 1;
+
+    //Arm
+    double liftPower = 1;
+    double shoulderPower = 1;
+    double elbowPower = 1;
 
     frontRight = hardwareMap.get(DcMotor.class, "frontRight");
```

Comment:
You still need to get the arm components from the hardware map. Without that the components just reference null

Reply:
I'm assuming you are referencing line 40. hardwareMap is added on line 9.

Reply:
You need to make sure to initialize the components new components added on lines 21-26 with hardwaremap.get. I.E
`hardwaremap.get(Servo.class, "wrist_servo")`
(the second value is arbitrary since we have the ability to configure it ourselves)
this needs to be done with each new component you added including the servo controller

Reply:
sorry for the unclear wording the in original review